cmake/Findfmt.cmake

```diff
@@ -0,0 +1,25 @@
+#[===[.md
+# Findfmt
+
+fmt compatibility module for Octopus
+
+This file is specifically tuned for Template usage to unify the api with upstream package.
+
+]===]
+
+list(APPEND CMAKE_MESSAGE_CONTEXT Findfmt)
+include(Template_Helpers)
+Template_FindPackage(${CMAKE_FIND_PACKAGE_NAME}
+		NAMES fmt
+		PKG_MODULE_NAMES fmt
+)
+
+# Create appropriate aliases
+if (${CMAKE_FIND_PACKAGE_NAME}_PKGCONFIG)
+	add_library(fmt::fmt ALIAS PkgConfig::${CMAKE_FIND_PACKAGE_NAME})
+endif ()
+set_package_properties(${CMAKE_FIND_PACKAGE_NAME} PROPERTIES
+		URL https://github.com/fmtlib/fmt
+		DESCRIPTION "A modern formatting library "
+)
```

Comment:
The goal of this `Findfmt.cmake` module is to make a unified interface of `fmt::fmt` regardless of if it is imported via `FetchContent` (outside of this file), or `find_package(CONFIG)` or `pkg-config` (handled in `Template_FindPackage`). This could also be expanded to handled un-packaged libraries, but that has to be defined manually